tests/test_helper.tcl

```diff
@@ -563,6 +565,7 @@ proc print_help_screen {} {
         "--port <port>      TCP port to use against external host."
         "--baseport <port>  Initial port number for spawned redis servers."
         "--portcount <num>  Port range for spawned redis servers."
+        "--packed-encoding <listpack|ziplist>  Default packed encoding."
```

Comment:
Using ```--run-on-startup``` causes test fail when require auth, so I changed ```set-default-packed-encoding``` to config, do you think this is appropriate?
Or should I only run ```--run-on-startup``` if not need to require auth?

Reply:
@oranagra I'm not sure need to add this debug trick, because hashZiplistValidateIntegrity will fail when full sanitization.
Are you worried that in the future we may default to full ziplist conversion to listpack resulting in corrupt-dump and fuzzy tests fail?

Reply:
i suppose we can make that `--run-on-startup` a best effort thing. if we get AUTH error, we ignore it.
for our use case, we add it so that we can run the entire test suite with some modification (on encoding), so it's not that bad if some test (specifically ones that test ACL) will not use it (won't affect our data type / encoding tests).

Another benefit of the `--run-on-startup`, is that it can be used together with `--host` on a preexisting server (where we don't control the config).

bottom line, i think i still prefer it because:
1. we don't promote something we wanted as a debug command into a config.
2. we create a generic mechanism that can be used for other things.

the downside is that this way we don't control creation of keys when loading rdb / aof file.
but the upside is that it is clear this isn't a feature of redis (it's a debug command, rather than an undocumented config), we won't have any concerns about breaking backwards compatibility if we ever delete it.